### ISSUE TITLE
[A11y] Onboarding Flow

### ISSUE THREAD

**Issue by rt4914** [oppia/oppia-android#2613]
## Current Output
https://user-images.githubusercontent.com/9396084/106880604-8f7f1d00-6702-11eb-81fd-975a8c4238ad.mp4

## Issues Identified
- [ ] Merge Title and Description w.r.t. Talkback
Talkback first outputs the heading "Welcome to Oppia!" and then leaner goes to next action and it outputs the subtitle "Learn anything you want in an effective and enjoyable way.", but instead it should be done in single action for better experience like "Welcome to Oppia! Learn anything you want in an effective and enjoyable way.".  

- [x] The bottom dots should read as progress like "Slide 1 of 4"
Talkback shifts from **Skip** to **Next** button directly. But instead I think it would be nice if talkback can output the dots as progress so that learner can understand how many slides are there and can decide to skip or not.

- [x] Need content description for **Next** arrow 
Currently talkback is unable to read the next arrow correctly and we need a correct content description for that. **Next**. Talkback will read this as **Next, Button** (@adit19shah)

- [x] Update overall talkback flow
Correct focus should be: `Heading and description → Next → Slide n of 4 → Skip` and repeat on next slide.

These issues are same for mobile-landscape, mobile-portrait, tablet-landscape and tablet-portrait.

Note: To understand the above issue(s) completely, it is recommended that you setup Talkback and play with the app keeping it on and that will give you better context.

**Accessibility Guide**: https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide

**Comment by rt4914** [comment]
Waiting for response on above blocked items from @mschanteltc

**Comment by mschanteltc** [comment]
- Let's merge the heading and description. It would be a better experience for both to be read at the same time because it takes a single action from the user. 
- I agree that the four dots should be read as "Slide 1 of 4" but add "double-tap to activate" if the user wants to navigate to a certain slide on the screen. Would this be possible?
- Yes "Next, Button" would work best.
- Let's have the focus be on the heading once the user navigates to the next page. However if we do this, can we rearrange the order as so? :
   - Heading and description → Next → Slide n of 4 → Skip
   - This way, the user doesn't have to swipe multiple times to use the "Next, Button"

**Comment by rt4914** [comment]
> * Let's merge the heading and description. It would be a better experience for both to be read at the same time because it takes a single action from the user.
> * I agree that the four dots should be read as "Slide 1 of 4" but add "double-tap to activate" if the user wants to navigate to a certain slide on the screen. Would this be possible?
> * Yes "Next, Button" would work best.
> * Let's have the focus be on the heading once the user navigates to the next page. However if we do this, can we rearrange the order as so? :
>   
>   * Heading and description → Next → Slide n of 4 → Skip
>   * This way, the user doesn't have to swipe multiple times to use the "Next, Button"

@mschanteltc Regarding: `I agree that the four dots should be read as "Slide 1 of 4" but add "double-tap to activate" if the user wants to navigate to a certain slide on the screen. Would this be possible?`
Making it clickable will create following issues:
(a.) Currently its not doable for regular learners too and therefore there will be a big inconsistency. Note: That left/right swipe of slides can be done by both types of learners.
(b.) If we make it clickable that means we will need to make each dot clickable to switch to any specific slide. Which in turn result in something like this for talkback: `Heading and description → Next → Slide 1 of 4, checked, double tap to activate → Slide 2 of 4, not-checked, double tap to activate → Slide 3 of 4, not-checked, double tap to activate → Slide 4 of 4, not-checked, double tap to activate → Skip ` . I think this will become too much options which can make things confusing.
(c.) Also, if we make items clickable then UI will also change because every clickable items needs be 48dpx48dp min size which means that the dots will look more separated.

WDYT?

**Comment by mschanteltc** [comment]
If the feature wasn't available when talkback was disabled, then let's scrap the idea of tapping to a certain slide altogether.

**Comment by rt4914** [comment]
> If the feature wasn't available when talkback was disabled, then let's scrap the idea of tapping to a certain slide altogether.

@mschanteltc Thanks. Updated.

**Comment by adit19shah** [comment]
@rt4914 I want to work on first and third part of this issue. i.e. "Merge Title and Description w.r.t. Talkback" and "Need Content Description for Next arrow".

**Comment by rt4914** [comment]
@adit19shah Assigned. Thanks

**Comment by anandwana001** [comment]
@yashraj-iitr are you still working on the due part of this issue?

**Comment by yashraj-01** [comment]
> @yashraj-iitr are you still working on the due part of this issue?

Actually, I have completed my part of the issue in PR #3105. @adit19shah was working on one of the parts so I did not work on that. Since there is no response from @adit19shah, should I work on that part of the issue or should I unassign myself?

**Comment by anandwana001** [comment]
If you work on it if you want.

**Comment by yashraj-01** [comment]
> If you work on it if you want.

Okay. I'll work on that part of the issue as well.
Also, @rt4914 I think the talkback flow part is not completed yet so you may want to uncheck it.

**Comment by rt4914** [comment]
> > If you work on it if you want.
> 
> Okay. I'll work on that part of the issue as well.
> Also, @rt4914 I think the talkback flow part is not completed yet so you may want to uncheck it.

@yashraj-iitr I discussed with the Ben and the final conclusion was that we should not write code to control the flow as its not a good practice and therefore we will be keeping the existing flow.

**Comment by rt4914** [comment]
@yashraj-iitr Adit is still working on this issue and therefore I am assigning it to him.
